nvdaHelper/remote/sysListView32.cpp

```diff
@@ -59,3 +59,46 @@ error_status_t nvdaInProcUtils_sysListView32_getColumnContent(handle_t bindingHa
 	*text = SysAllocString(lvItem.pszText);
 	return 0;
 }
+
+error_status_t nvdaInProcUtils_sysListView32_getColumnLocation(handle_t bindingHandle, const unsigned long windowHandle, int item, int subItem, RECT* location) {
+	RECT localRect {
+		// Returns the bounding rectangle of the entire item, including the icon and label.
+		LVIR_LABEL,  // left
+		// According to Microsoft, top should be the one-based index of the subitem.
+		// However, indexes coming from LVM_GETCOLUMNORDERARRAY are zero based.
+		subItem // top
+	};
+	if (!SendMessage((HWND)UlongToHandle(windowHandle), LVM_GETSUBITEMRECT, (WPARAM)item, (LPARAM)&localRect)) {
+		LOG_DEBUGWARNING(L"LVM_GETSUBITEMRECT failed");
+		return 1;
+	}
+	*location = localRect;

```

Comment:
We use an internal rect struct in SendMessage since it needs to be initialized properly. We shouldn't do this in the structure provided by the caller IMO.

Reply:
Could you expand on why we shouldn't?

On success we modify the callers structure, that is a requirement. I understand we may not wish to do so if the call fails, therefore use a temporary until we can verify success, then copy. Another reason is just because it is easier to initialize a fresh struct, than zero out and initialize an existing one.

Reply:
Actually both the reasons you mention are valid, but I think the former is most important. I'll document it.